### PR TITLE
refactor(process): process units return FluidStream | PropagationFailure

### PR DESCRIPTION
**Pull request by olelod**
## Type of Work

- [x] Patch: X.Y.**Z+1**. **NEGLIGIBLE** visible changes, does not change input or output - OR changes behaviour. Use chore:, refactor: etc
- [ ] Minor: X.**Y+1**.Z. Minor changes, might ADD new input (YAML), or other **backwards-compatible** changes. Use feat:, fix:
- [ ] Major: **X+1**.Y.Z. Major and most likely **BREAKING** changes, wo. backwards compatibility, or removing temporary backwards compatibility functionality. Use ! or BREAKING:.

See here (internal): https://github.com/equinor/ecalc-internal/discussions/1044

## Have you remembered and considered?

- [ ] IF FEAT: I have remembered to update documentation
- [ ] IF FIX OR FEAT: I have remembered to update manual changelog (`docs/drafts/next.draft.md`)
- [ ] IF BREAKING: I have remembered to update migration guide (`docs/docs/migration_guides/`)
- [ ] IF BREAKING: I have committed with `BREAKING:` in footer or `!` in header
- [x] I have added tests (if not, comment why) — existing tests that asserted on raised exceptions were ported to assert on returned `PropagationFailure` variants instead; no new coverage added because no new behaviour was introduced
- [x] I have used conventional commits syntax (if you squash, make sure that conventional commit is used)
- [ ] I have included the Github issue nr in the footer!

## What is this PR all about?

Process units no longer raise exceptions when they can't propagate a stream — they return a typed `PropagationFailure` value instead.

**Signature change.** Every concrete `ProcessUnit.propagate_stream(...)` now returns:

```python
FluidStream | PropagationFailure
```

instead of returning `FluidStream` and raising on failure.

**Deletions.** `process_pipeline/process_error.py` is removed. The three exception classes it held — `RateTooHighError`, `RateTooLowError`, `OutsideOperationalCapacityError` — are gone. Each had a one-to-one counterpart variant on `PropagationFailure` introduced in PR 1; the migration is mechanical.

**Call-site updates.**
- `stream_propagator` forwards the union.
- Solvers (`speed`, `recirculation`, `upstream_choke`, `downstream_choke`, `multi_pressure`, `multi_shaft`, `outlet_pressure`, `feasibility`) drop `try/except Rate*Error` blocks in favour of `is_failure(...)` / `isinstance(..., PropagationFailure)` branches, and wrap the result in `Solution.failed(...)`.
- Pressure-control strategies (`common_asv`, `individual_asv`, `upstream_choke`, `downstream_choke`) do the same. `IndividualASVRateControlStrategy` keeps a private `_InfeasibleDuringSearch` exception **scoped inside** `get_outlet_pressure` to short-circuit `find_root` mid-iteration when a propagated failure makes continuing pointless; it never escapes the method.
- Tests that previously used `pytest.raises(Rate*Error)` were ported to assert on returned variants.

**No behavioural change.** Same inputs produce the same outputs, including the same failure outcomes. The change is in *how* the failure is delivered (return value vs. raised exception), not *what* the failure is.

## Between the lines?

**Stack**

1. [#1578](https://github.com/equinor/ecalc/pull/1578) — rename `SolverFailure` → `PropagationFailure`.
2. **This PR** — process units return `FluidStream | PropagationFailure`; unit exception classes deleted.
3. `typed-failures/03-search-strategies` — add `DidNotConverge` variant; search/root-finding return `float | DidNotConverge`; `DidNotConvergeError` deleted.

### CODE REVIEW

File: src/libecalc/domain/process/compressor/core/train/stage.py [hunk 1]
```diff
@@ -130,7 +131,11 @@ def get_compressor_inlet_stream(
 
         # Then the stream passes through the PressureModifier (if defined),
         if self.choke is not None:
-            inlet_stream_after_choke = self.choke.propagate_stream(inlet_stream_after_mixer)
+            choke_result = self.choke.propagate_stream(inlet_stream_after_mixer)
+            if isinstance(choke_result, PropagationFailure):
+                # Legacy compressor train code has no PropagationFailure channel; surface as an exception
+                raise IllegalStateException(f"Choke could not propagate stream: {choke_result}")
```
Comment (tj098895) [inline src/libecalc/domain/process/compressor/core/train/stage.py:137]: I think we created a new `IllegalStateException` class called `IllegalProcessStreamStateException` or sth, as the `IllegalStateException` is imo sth related to programming errors

Comment (tj098895) [reply src/libecalc/domain/process/compressor/core/train/stage.py:137]: Also, it should send choke_result as a separate parameter instead of serializing it here.

Comment (tj098895) [reply src/libecalc/domain/process/compressor/core/train/stage.py:137]: also include process_unit_id ?

File: src/libecalc/process/process_solver/solvers/recirculation_solver.py [hunk 2]
```diff
@@ -33,79 +36,103 @@ def bool_func(x: float, mode: Literal["minimize", "maximize"]) -> tuple[bool, bo
...
             """
-            try:
-                func(RecirculationConfiguration(recirculation_rate=x))
-                return False if mode == "minimize" else True, True
-            except RateTooLowError:
+            result = func(RecirculationConfiguration(recirculation_rate=x))
+            if isinstance(result, RateTooLow):
                 return True, False
-            except RateTooHighError:
+            if isinstance(result, RateTooHigh):
```
Comment (tj098895) [inline src/libecalc/process/process_solver/solvers/recirculation_solver.py:42]: Rename `RateTooHigh` to `RateTooHighStreamErrorEvent` or similar? We need one error representing all the states/errors that can occur with a stream - ie violating policies of a process_unit;

- Rate too high/low
- Pressure too high/low
- Temperature too high/low
- Illegal composition? Is that possible? Incompatible composition? ie that fluid reaches a compressor?

Imo. every process unit should have policies against these, ie. need to make a decision on what are the limits, either hard limits or limits calculated such as compressor chart. We need to make sure that every process unit makes a decision on what do to in those cases, and that we know that all edge cases are checked and taken care of ... possibly enums of PropagationFailure, in order for a solver to make decisions on what to test next or to give up?

File: src/libecalc/domain/process/compressor/core/train/stage.py (hunk 1 above)
Comment (olelod) [reply src/libecalc/domain/process/compressor/core/train/stage.py:137]: this is the legacy compressor stage
